orttraining/orttraining/python/training/optim/fused_adam.py

```diff
@@ -80,7 +80,10 @@ def __init__(self,
         # Skip buffer
         self._dummy_overflow_buf = torch.cuda.IntTensor([0])
 
-        from onnxruntime.training.ortmodule.torch_cpp_extensions import fused_ops
+        try:
+            from onnxruntime.training.ortmodule.torch_cpp_extensions import fused_ops
+        except ImportError:
+            import fused_ops
```

Comment:
Can this be a separate function?

Reply:
Good point. I was drafting a poc to refactor like this using https://docs.python.org/3/library/importlib.html#importlib.import_module

Something like
```
def import_from_onnxruntime_or_environment(name):
    ortmodule_path='onnxruntime.training.ortmodule.torch_cpp_extensions'
    try:
        import importlib
        mylib = importlib.import_module('.', ortmodule_path+'.'+name)
    except ModuleNotFoundError:
        print(name)
        mylib = importlib.import_module('.', name)
```

But although it imported the module, it wasn't defined. Some more coding must be needed to achieve this nicer refactoring

Reply:
Why not something like:

```
def get_fused_ops_extension():
        try:
            from onnxruntime.training.ortmodule.torch_cpp_extensions import fused_ops
        except ImportError:
            # print, warnings, logging?
            import fused_ops
        return fused_ops

fused_ops = get_fused_ops_extension()
```

I had issues compiling these extensions on Windows (long path + unable to import them when copied at the expected location), and on Linux CPU (it complains about CUDA home if it is compiled with install.py).